releasenotes/notes/43424.yaml

```
@@ -0,0 +1,7 @@
+apiVersion: release-notes/v2
+kind: feature
+area: security
+
+releaseNotes:
+  - |
+    **Removed** `PILOT_SKIP_VALIDATE_TRUST_DOMAIN` env from pilot which was unused for a long time.
```

```suggestion
    **Removed** `PILOT_SKIP_VALIDATE_TRUST_DOMAIN` environment variable from Pilot.
```

Also recommend an alternative suggestion.